src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/custom.py

```diff
@@ -139,6 +139,8 @@ def _add_whl_ext(source, ext_sha256=None, extra_index_urls=None):  # pylint: dis
     extension_path = get_extension_path(extension_name)
     pip_args = ['install', '--target', extension_path, ext_file]
 
+    if pip_proxy:
+        pip_args = ['--proxy', pip_proxy] + pip_args
```

Comment:
Why is this added before `install --target etc.`?
Does it not work if you just append to the end of `pip_args`?

Reply:
The reason is that the --proxy param is an option of base [pip](https://pip.pypa.io/en/stable/reference/pip/) rather than the [install command](https://pip.pypa.io/en/stable/reference/pip_install/) so opt-ed for this way to be 'closer to documentation'. There are plenty of examples in the other way so I don't mind changing.